guide/content/en/guide/basics/handlers.md

```diff
@@ -129,3 +140,52 @@ from sanic.request import Request
 async def typed_handler(request: Request) -> HTTPResponse:
     return text("Done.")
 ```
+
+## Naming your handlers
+
+All handlers are named automatically. This is useful for debugging, and for generating URLs in templates. When not specified, the name that will be used is the name of the function.
+
+.. column::
+
+    For example, this handler will be named `foo_handler`.
+
+.. column::
+
+    ```python
+    # Handler name will be "foo_handler"
+    @app.get("/foo")
+    async def foo_handler(request):
+        return text("I said foo!")
+    ```
+
+.. column::
+
+    However, you can override this by passing the `name` argument to the decorator.
+
+.. column::
+
+    ```python
+    # Handler name will be "foo"
+    @app.get("/foo", name="foo")
+    async def foo_handler(request):
+        return text("I said foo!")
+    ```
+
+.. column::
+
+    In fact, as you will, there may be times when you **MUST** supply a name. For example, if you use two decorators on the same function, you will need to supply a name for at least one of them.
+    
+    If you do not, you will get an error and your app will not start. Names **must** be unique within your app.
+
+.. column::
+
+    ```python
+    # Two handlers, same function,
+    # different names:
+    # - "foo"
+    # - "foo_handler"
+    @app.get("/foo", name="foo")
+    @app.get("/bar")
+    async def foo_handler(request):
+        return text("I said foo!")
+    ```
```

Comment:
```suggestion
    ```python
    # Two handlers, same function,
    # different names:
    # - "foo"
    # - "foo_handler"
    @app.get("/foo/<arg>", name="foo_arg")
    @app.get("/foo")
    async def foo(request, arg=None):
        return text("I said foo!")
    ```
```
More sensible naming, a common use case with default arguments. If you don't want to get to this yet, at least avoid name `foo_handler` matching `/bar` which is just confusing. Possibly return `url_for` in the message?